src/app/pages/permissions-page/permissions.page.ts

```diff
@@ -22,66 +25,120 @@ export class PermissionsPage {
 
   constructor( private router: Router,
     private _authService: AuthService,
+    private _alertService: AlertService,
     private _permissionsModelService: PermissionsModelService,
     private _loadingService: LoadingService) {
 
   }
 
-  
+
 
   ionViewWillEnter() {
-    this._loadingService.show({message: "..loading.."}).then(() => { 
+    this._loadingService.show({message: "..loading.."}).then(() => {
       this._permissionsModelService.init();
       this._loadingService.dismiss ();
     })
   }
 
-  hasUser = false;
-
-  selectedSkills = [];
-
-  theCurrentUser = this.getUser();
 
-  selectedUser = {};
+  selectedRole: string = '';
 
-  
-  
   selectUser(user) {
-    let role = [];
-    for (let i of user.roles){
-    role.push(i.name);
+    if (this._permissionsModelService.isDirty() == false) {
+      this._permissionsModelService.selectedUserRoles = [];
+      for (let i of user.roles){
+      this._permissionsModelService.selectedUserRoles.push(i.name);
+      }
+      this._permissionsModelService.selectedUserName = user.name;
+      this._permissionsModelService.hasSelectedUser = true;
     }
-    this.selectedUser = user;
-    this.selectedSkills = role;
-    this.hasUser = true;
+    else {
+      this._alertService.show({
+        header: 'Changes Not Saved!',
+        message: 'Discard changes?',
+        buttons: [{
+          text: "Go Back",
+          role: 'cancel'
+        }, {
+          text: "Discard" ,
+          handler: () => {
+            this._permissionsModelService.dirty = false;
+            this.selectUser(user);
+          }
+        }]
+      })
+    }
+  }
+
+
+  hasSelectedUser() {
+    return this._permissionsModelService.hasSelectedUser
+  } 
+
+  getselectedUserRoles() {
+    return this._permissionsModelService.selectedUserRoles;
   }
-  
-  clearUser() {
-    this.selectedSkills = [];
-    this.hasUser = false;
-    
+
+  getSelectedUserName() {
+    return this._permissionsModelService.selectedUserName;
   }
 
-  getUser() {
-    return this._authService.getUser();
+  getCurrentUserName() {
+    return this._authService.getUser().name;
   }
-  
-  
+
+
   getListOfUsers() {
     return this._permissionsModelService.getListOfUsers();
   }
 
+
+
   getListOfRoles() {
-    return this._permissionsModelService.getListOfRoles();
+    let availableRoles = [];
+    let allroles = this._permissionsModelService.getListOfRoles();
+    if (allroles !== undefined && allroles !== null) {
+      availableRoles = [];
+      for (let j of Object.values(allroles)) {
+        if (this._permissionsModelService.selectedUserRoles.includes( j['name'])) {
+          continue;
```

Comment:
No need to use `continue`. Think about the `if` statement another way. As you have it, it is `if (true) continue else { do something }`. Write it as `if (not true)` .. or `if (false)` then `do something`. That way the `continue` is not necessary. You will rarely if ever need `continue`, maaayybe `break` occasionally, but rarely if ever `continue`.